src/pages/CustomerRequisitionsPage.js

```diff
@@ -1,149 +1,124 @@
+  const [state, dispatch, debouncedDispatch] = usePageReducer(routeName, {}, initialiseState);
 
-    navigateTo('customerRequisition', `${navStrings.requisition} ${requisition.serialNumber}`, {
-      requisition,
-    });
-  };
+  const { data, sortBy, isAscending, searchTerm, PageActions, columns, keyExtractor } = state;
 
-  updateDataFilters = (newSearchTerm, newSortBy, newIsAscending) => {
-    // (... != null) checks for null or undefined (implicitly type coerced to null).
-    if (newSearchTerm != null) this.dataFilters.searchTerm = newSearchTerm;
-    if (newSortBy != null) this.dataFilters.sortBy = newSortBy;
-    if (newIsAscending != null) this.dataFilters.isAscending = newIsAscending;
-  };
+  const refreshCallback = () => dispatch(PageActions.refreshData(), []);
+  // Custom hook to refresh data on this page when becoming the head of the stack again.
+  useNavigationFocus(refreshCallback, navigation);
+  // Custom hook to listen to sync changes - refreshing data when requisitions are synced.
+  useSyncListener(refreshCallback, 'Requisition');
 
-  /**
-   * Returns updated data filtered by |searchTerm| and ordered by |sortBy| and |isAscending|.
-   */
-  refreshData = (newSearchTerm, newSortBy, newIsAscending) => {
-    this.updateDataFilters(newSearchTerm, newSortBy, newIsAscending);
-    const { searchTerm, sortBy, isAscending } = this.dataFilters;
-    const data = this.requisitions.filtered('serialNumber BEGINSWITH $0', searchTerm);
-    let sortDataType;
-    switch (sortBy) {
-      case 'serialNumber':
-      case 'numberOfItems':
-        sortDataType = 'number';
-        break;
-      default:
-        sortDataType = 'realm';
-    }
-    this.setState({
-      data: sortDataBy(data, sortBy, sortDataType, isAscending),
-    });
-  };
+  const onPressRow = useCallback(rowData => reduxDispatch(gotoCustomerRequisition(rowData)), []);
+  const onSearchFiltering = value => dispatch(PageActions.filterData(value));
 
-  renderCell = (key, requisition) => {
-    switch (key) {
-      case 'entryDate':
-        return requisition.entryDate.toDateString();
-      case 'customerName':
-        return requisition.otherStoreName ? requisition.otherStoreName.name : '';
-      case 'status':
-        return formatStatus(requisition.status);
-      default:
-        return requisition[key];
-    }
-  };
+  const renderRow = useCallback(
+    listItem => {
+      const { item, index } = listItem;
+      const rowKey = keyExtractor(item);
+      return (
+        <DataTableRow
+          rowData={data[index]}
+          rowKey={rowKey}
+          columns={columns}
+          dispatch={dispatch}
+          onPress={onPressRow}
+          rowIndex={index}
+        />
+      );
+    },
+    [data]
+  );
 
-  render() {
-    const { data } = this.state;
-    const { database, genericTablePageStyles, topRoute } = this.props;
+  const renderHeader = useCallback(
```

Comment:
this block appears in a number of components, is it possible to move the `useCallback` into the `DataTableHeadRow` component?